### PR TITLE
ENGG-1738: Integrate RequestBot

### PR DESCRIPTION
**Pull request by RuntimeTerror10**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/requestBot/RequestBot.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <m.div
+      initial={{ opacity: 0, right: -50 }}
+      animate={{ opacity: isOpen ? 1 : 0, right: isOpen ? 0 : -50 }}
+      transition={{ duration: 0.2 }}
+      className="request-bot-container"
+    >
+      <div className="request-bot">
+        <IoMdClose className="request-bot-close-btn" onClick={onClose} />
+        <iframe
+          frameBorder="0"
```
Comment (nsrCodes) [inline app/src/features/requestBot/RequestBot.tsx:22]: do this via css

Comment (nsrCodes) [reply app/src/features/requestBot/RequestBot.tsx:22]: also, add a title to the iframe

File: app/src/layouts/DashboardLayout/MenuHeader/index.js [hunk 2]
```diff
@@ -105,38 +130,59 @@ const MenuHeader = () => {
...
+                  className="ask-ai-btn"
+                  onClick={() => {
+                    setIsRequestBotVisible(true);
+                    dispatch(
+                      actions.updateProductTourCompleted({
+                        tour: TOUR_TYPES.MISCELLANEOUS,
+                        subTour: "askAI",
+                      })
+                    );
+                  }}
```
Comment (nsrCodes) [inline app/src/layouts/DashboardLayout/MenuHeader/index.js:144]: add analytics events here

**Review by nsrCodes — APPROVED** [review]
LGTM, just minor nit-picks. Make sure to address them before merging.

Also, if possible address [this comment by deepsource](https://app.deepsource.com/gh/requestly/requestly/run/d69d75a5-b559-4d95-b2d6-cc11e930e3f5/javascript/JS-D010?listindex=7)(_and any other security mesaures about iframes_) as it otherwise _might_ comeup during security scans

File: app/src/features/requestBot/RequestBot.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          src="https://widget.writesonic.com/CDN/index.html?service-base-url=https%3A%2F%2Fapi-azure.botsonic.ai&token=ecb64aff-5d8a-40e6-b07b-80b14997c80f&base-origin=https%3A%2F%2Fbot.writesonic.com&instance-name=Botsonic&standalone=true&page-url=https%3A%2F%2Fbot.writesonic.com%2Fbots%2Fd59d951e-714f-41d9-8834-4d8dfa437b0e%2Fintegrations"
+        ></iframe>
+        <div
+          className="get-human-support"
+          onClick={() => {
+            onClose();
+            window.$saturn.open();
```
Comment (rohanmathur91) [inline app/src/features/requestBot/RequestBot.tsx:29]:
```suggestion
            window.$saturn?.open();
```

**Pull request merged**